src/index.tsx

```diff
@@ -79,35 +77,34 @@ export function useSpeechRecognition(options: SpeechRecognitionOptions = default
 
   const resetTranscript = useCallback(() => {
     disconnect(SpeechRecognitionDisconnectType.RESET);
+    dispatch(setError(undefined));
     dispatch(setTranscript(""));
     dispatch(setInterimTranscript(""));
     dispatch(setFinalTranscript(""));
   }, [disconnect]);
 
   const startListening = useCallback(() => {
-    if (recognition && !listening) {
+    if (recognition && status !== SpeechRecognitionStatus.STARTED) {
       if (!recognition.continuous) {
         resetTranscript();
       }
+
       try {
         recognition.start();
+        dispatch(setStatus(SpeechRecognitionStatus.STARTED));
       } catch (DOMException) {
         // Tried to start recognition after it has already started - safe to swallow this error
       }
-      dispatch(setStatus(SpeechRecognitionStatus.STARTED));
-      dispatch(setListening(true));
     }
-  }, [listening, recognition, resetTranscript]);
+  }, [status, recognition, resetTranscript]);
 
   const stopListening = useCallback(() => {
     disconnect(SpeechRecognitionDisconnectType.STOP);
     dispatch(setStatus(SpeechRecognitionStatus.STOPPED));
```

Comment:
this status is dispatched in the `disconnect` function above.  Can probably safely just call `disconnect(SpeechRecognitionDisconnectType.STOP);`